### PR TITLE
vmware: add tectonic_vmware_resource_pool variable

### PR DESCRIPTION
**Pull request by rraca**
Adds possibility to start vm's in resource pool

### CODE REVIEW

File: platforms/vmware/variables.tf [hunk 1]
```diff
@@ -45,6 +45,11 @@ variable "tectonic_vmware_cluster" {
   description = "vCenter Cluster used to create VMs under"
 }
 
+variable "tectonic_vmware_resource_pool" {
+  type        = "string"
+  description = "The name of a Resource Pool in which to launch the virtual machine. Requires full path"
+}
+
```
Comment (alekssaul) [inline platforms/vmware/variables.tf]: Can we setup a `default = ""`, haven't tested but I believe that should let it create the VMs without adding it to a resource pool.

File: examples/terraform.tfvars.vmware [hunk 2]
```diff
@@ -102,6 +102,9 @@ tectonic_vanilla_k8s = false
 // vCenter Cluster used to create VMs under
 tectonic_vmware_cluster = ""
 
+// (optional) The name of a Resource Pool in which to launch the virtual machine. Requires full path
+tectonic_vmware_resource_pool = ""
+
```
Comment (alekssaul) [inline examples/terraform.tfvars.vmware]: Was this edited by hand? we generate these programmatically using `make examples` . You should be able to keep this same as variables.tf and run `make examples` across the repo.

**Comment by alekssaul** [comment]
hey @rraca  , thank you for your contribution!!

As a general measure please make sure to run make examples docs and make structure-check before submitting PRs to tectonic-installer repo. terraform fmt . on root folder should handle the formatting issues. This has been why most of my builds fail :)

Made quick comments in code that I am hoping makes sense. I am hoping to test this with/without RPs in the cluster in the next couple days.

**Comment by rraca** [comment]
Hi @alekssaul 
I'v added default to variable and did all makes :)

**Comment by alekssaul** [comment]
Thanks @rraca , changes looks good to me. Unfortunately, we don't yet have automated tests for VMware environments. I'd like to test this with/without RPs in the next couple days.

**Comment by alekssaul** [comment]
Hi @rraca ,

Thanks again for contributing. This is the first PR for VMware platform from the community :1st_place_medal: 

I've tested the PR functionally by:
- Running terraform apply without specifying `tectonic_vmware_resource_pool` variable. All VMs are created under the cluster outside of Resource Pool
- Running terraform apply with `tectonic_vmware_resource_pool = /nyc/host/cluster01/Resources/rp` in my test cluster, which creates the VMs inside "rp" resource pool under "cluster01" in "NYC" datacenter.

I made a small addition to variable description (Full path can be obtained by executing `govc pool.info $ResourcePoolName`) and submitted a PR into your branch. If it looks, feel free to merge it to your branch. In either case, this looks good to me.

**Comment by alekssaul** [comment]
Hey @s-urbaniak , can we get this in if no concerns?

**Comment by s-urbaniak** [comment]
ok to test

**Comment by s-urbaniak** [comment]
@alekssaul No concerns from my side once green

**Comment by s-urbaniak** [comment]
@rraca Thanks a lot for the contribution! 👍

**Comment by kongslund** [comment]
Any chance this PR will be merged soon?

**Comment by alekssaul** [comment]
Hi @kongslund , we are in the process of cutting a release and the our engineers are a bit tied down. We would like to merge this soon, however realistically it may need to wait after the release is cut. 

Apologies for the delay and thank you for your patience.

**Comment by alekssaul** [comment]
ping @s-urbaniak : If this generally look good to you, I think it would make a good contribution. Happy to test manually if it looks reasonable. It looked good to me ~couple weeks ago..

**Comment by metral** [comment]
@s-urbaniak Do we have any updates on this PR?

**Comment by s-urbaniak** [comment]
@metral @alekssaul no concerns from my side, it just needs a rebase.

**Comment by cpanato** [comment]
@rraca Thanks for the PR, could you please rebase when you have some time?

**Comment by cpanato** [comment]
Hello @rraca can you please rebase and fix the conflicts? thanks!

**Comment by cpanato** [comment]
closing due to inactivity. please reopen another PR if you want this.

**Pull request closed without merging**